File: BlogApp-RN/src/features/posts/domain/usecases/addPost.ts
import { AsyncResult } from "../../../../core/typedefs/asyncResult";
import { CompletedPost } from "../../data/models/completedPost";
import { PostTemplate } from "../../data/models/postTemplate";
import PostRepo from "../repos/postRepo";

class AddPost {
  constructor(private repo: PostRepo) {}
  call: (post: PostTemplate) => AsyncResult<CompletedPost> = (post) => {
    return this.repo.addPost(post);
  };
}

export default AddPost;

File: blogapp_flutter/lib/features/posts/domain/usecases/add_post.dart
import 'package:blogapp_flutter/core/typedefs/async_result.dart';
import 'package:blogapp_flutter/core/typedefs/usecases.dart';
import 'package:blogapp_flutter/features/posts/data/models/completed_post.dart';
import 'package:blogapp_flutter/features/posts/data/models/post_template.dart';
import 'package:blogapp_flutter/features/posts/domain/repos/posts_repo.dart';

class AddPost extends UsecaseWithParams<CompletedPost, PostTemplate> {
  final PostsRepo _repo;
  AddPost(this._repo);
  @override
  AsyncResult<CompletedPost> call(params) {
    return _repo.addPost(params);
  }
}
